### PR TITLE
Only pass non-empty parallel labels

### PR DESCRIPTION
**Pull request by streeve**
~~Deprecate~~ Only forward labels with `simd_parallel_for` and `neighbor_parallel_for` if not empty ~~and add required label versions~~. This was discussed in #183

### CODE REVIEW

**Comment by sslattery** [comment]
> Did you consider only passing the label if it is not empty instead of requiring it?

What is the current Kokkos strategy for labels? Are they always required now?

**Comment by dalg24** [comment]
> > Did you consider only passing the label if it is not empty instead of requiring it?
> 
> What is the current Kokkos strategy for labels? Are they always required now?

No

**Comment by streeve** [comment]
> Did you consider only passing the label if it is not empty instead of requiring it?

Good point. I can do that - is it worth keeping the interface with name first?

File: core/src/Cabana_Parallel.hpp [hunk 1]
```diff
@@ -270,23 +281,27 @@ inline void neighbor_parallel_for(
...
+            for ( index_type a = n + 1; a < nn; ++a )
+            {
+                const index_type k =
+                    neighbor_list_traits::getNeighbor( list, i, a );
+                Impl::functorTagDispatch<work_tag>( functor, i, j, k );
             }
-        } );
+        }
+    };
+    if ( str == "" )
```
Comment (sslattery) [inline core/src/Cabana_Parallel.hpp:301]: Use `str.empty()`

**Comment by streeve** [comment]
Added `str.empty()` and rebased on master

**Pull request merged**